dynamiq/utils/utils.py

```diff
@@ -116,35 +119,80 @@ def format_value(value: Any, skip_format_types: set = None, force_format_types:
     if force_format_types is None:
         force_format_types = set()
 
+    metadata = {}
     if not isinstance(value, tuple(force_format_types)) and isinstance(
         value, tuple(skip_format_types)
     ):
-        return value
+        return value, metadata
 
     if isinstance(value, BytesIO):
-        return getattr(value, "name", None) or encode_bytes(value.getvalue())
+        return getattr(value, "name", None) or encode_bytes(value.getvalue()), metadata
     if isinstance(value, bytes):
-        return encode_bytes(value)
+        return encode_bytes(value), metadata
+
+    path = kwargs.get("path", "")
     if isinstance(value, dict):
-        return {
-            k: format_value(v, skip_format_types, force_format_types)
-            for k, v in value.items()
-        }
+        formatted_dict = {}
+        for k, v in value.items():
+            new_path = f"{path}.{k}" if path else k
+            formatted_v, sub_metadata = format_value(
+                v, skip_format_types, force_format_types, truncate_enabled, path=new_path
+            )
+            formatted_dict[k] = formatted_v
+            metadata.update(sub_metadata)
+        return formatted_dict, metadata
+
+    if truncate_enabled and isinstance(value, list) and all(isinstance(v, float) for v in value):
+        original_length = len(value)
+        if original_length > 20:
```

Comment:
Let's move this to constants and add to this method `truncate_limit: int = TRUNCATE_LIMIT`. So we can configure and change per each format if need.